<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GenCppSettings</class>
 <widget class="QDialog" name="GenCppSettings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>561</width>
    <height>443</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Параметры</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../res/fmt.qrc">
    <normaloff>:/img/Properties.png</normaloff>:/img/Properties.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_7">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Структуры и перечисления</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QGroupBox" name="structGroupBox">
         <property name="title">
          <string>Генерировать структуру</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <property name="leftMargin">
           <number>20</number>
          </property>
          <item>
           <widget class="QCheckBox" name="alignNamesStruct">
            <property name="text">
             <string>Выравнивать наименования</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="structComments">
            <property name="text">
             <string>Добавлять комментарии к полям</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="unionBox">
         <property name="title">
          <string>Объединения ключей</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QCheckBox" name="uAlignNamesStruct">
              <property name="text">
               <string>Выравнивать наименования</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Имя объеденения:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="unionNameMask">
              <property name="text">
               <string>${StructName}_KEYS</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QFrame" name="frame">
            <property name="frameShape">
             <enum>QFrame::HLine</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Комментарии:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout">
                <item>
                 <widget class="QRadioButton" name="unionNoComment">
                  <property name="text">
                   <string>Не создавать</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QRadioButton" name="unionCommentFld">
                  <property name="text">
                   <string>Комментарий поля</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QRadioButton" name="unionCommentSeg">
                <property name="text">
                 <string>Комментарий сегмента (если есть)</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1">
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Имена сегментов: </string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="segmentNameType">
              <item>
               <property name="text">
                <string>Префикс k и номер индекса (k1, k2, ...)</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Из наименований полей</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="enumGroupBox">
         <property name="title">
          <string>Перечесления</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Имя перечесления:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Имена сегментов: </string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QLineEdit" name="enumNameMask">
              <property name="text">
               <string>${StructName}_KEYNUM</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="enumNameType">
              <item>
               <property name="text">
                <string>Префикс имя структуры и номер индекса</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Префикс имя структуры и строка из наименований полей</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Функции</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_11">
       <item>
        <widget class="QCheckBox" name="genBTRVFILE">
         <property name="text">
          <string>Создавать BTRVFILE</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="genOpener">
         <property name="text">
          <string>Создавать функцию открытия BTRVFILE</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="genFindFunctions">
         <property name="title">
          <string>Создавать функции поиска</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_9">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="0,1">
            <item>
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Имена функций поиска:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="ffNameType">
              <item>
               <property name="text">
                <string>Нумерация</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Из наименования сегментов</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Функция поиска по умолчанию:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QRadioButton" name="ffDefPkKey">
                <property name="text">
                 <string>По первичном ключу</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="ffDefAinc">
                <property name="text">
                 <string>Из автоинкриментного индекса 
(если есть)</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="skfGroupBox">
         <property name="title">
          <string>Создавать SKF</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_10">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_9">
            <item>
             <widget class="QCheckBox" name="skfUniqKeys">
              <property name="text">
               <string>На каждый неуникальный ключ</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="skfDefault">
              <property name="text">
               <string>Один общий SKF</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8" stretch="0,1">
            <item>
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>Имена SKF функций:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="skfNameType">
              <item>
               <property name="text">
                <string>Нумерация</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Из наименования сегментов</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../res/fmt.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>GenCppSettings</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>GenCppSettings</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
